File: sdk/swift/Sources/AlignTechFeatureSDK/AlignTechClient.swift
import Foundation

public actor AlignTechClient {
    private let store: ConfigStore
    private let evaluator: Evaluator
    private let poller: ConfigPoller
    
    public init(config: AlignTechConfig) throws {
        guard !config.appKey.isEmpty else {
            throw AlignTechError.invalidConfig("appKey is required")
        }
        
        self.store = ConfigStore()
        self.evaluator = Evaluator()
        self.poller = ConfigPoller(config: config, store: store)
    }
    
    public func initialize() async {
        await poller.fetchInitial()
        await poller.start()
    }
    
    public func isEnabled(flagKey: String, user: AlignTechUser) async -> Bool {
        guard let flag = await store.get(flagKey) else {
            return false
        }
        return evaluator.evaluate(flag: flag, user: user).enabled
    }
    
    public func evaluate(flagKey: String, user: AlignTechUser) async -> EvaluationResult? {
        guard let flag = await store.get(flagKey) else {
            return nil
        }
        return evaluator.evaluate(flag: flag, user: user)
    }
    
    public func isReady() async -> Bool {
        return await !store.isEmpty()
    }
    
    public func setTraceParent(_ traceParent: String) async {
        await poller.setTraceParent(traceParent)
    }
    
    public func close() async {
        await poller.stop()
    }
}

public enum AlignTechError: Error {
    case invalidConfig(String)
    case networkError(String)
}

File: sdk/javascript/src/AlignTechClient.ts
import { AlignTechConfig, AlignTechUser, EvaluationResult } from './types';
import { ConfigStore } from './ConfigStore';
import { Evaluator } from './Evaluator';
import { ConfigPoller } from './ConfigPoller';

export class AlignTechClient {
  private store: ConfigStore;
  private evaluator: Evaluator;
  private poller: ConfigPoller;

  constructor(config: AlignTechConfig) {
    if (!config.appKey) {
      throw new Error('appKey is required');
    }

    this.store = new ConfigStore();
    this.evaluator = new Evaluator();
    this.poller = new ConfigPoller(config, this.store);
  }

  async initialize(): Promise<void> {
    await this.poller.fetchInitial();
    this.poller.start();
  }

  isEnabled(flagKey: string, user: AlignTechUser): boolean {
    const flag = this.store.get(flagKey);
    if (!flag) return false;
    return this.evaluator.evaluate(flag, user).enabled;
  }

  evaluate(flagKey: string, user: AlignTechUser): EvaluationResult | null {
    const flag = this.store.get(flagKey);
    if (!flag) return null;
    return this.evaluator.evaluate(flag, user);
  }

  isReady(): boolean {
    return !this.store.isEmpty();
  }

  setTraceParent(traceParent: string): void {
    this.poller.setTraceParent(traceParent);
  }

  close(): void {
    this.poller.stop();
  }
}
